TestInspectCmd fails a lot

```
--- FAIL: TestInspectCmd (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x2e7b5e3]

goroutine 9 [running]:
testing.tRunner.func1.2({0x367f020, 0x560abe0})
	/opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1734 +0x21c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1737 +0x35e
panic({0x367f020?, 0x560abe0?})
	/opt/hostedtoolcache/go/1.24.4/x64/src/runtime/panic.go:792 +0x132
github.com/inspektor-gadget/inspektor-gadget/pkg/params.(*Param).AsString(...)
	/home/runner/work/inspektor-gadget/inspektor-gadget/pkg/params/params.go:488
github.com/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler.(*OciHandlerInstance).init(0xc000488600, {0x3f20360, 0xc0000deb00})
	/home/runner/work/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler/oci.go:332 +0x1a3
github.com/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler.(*ociHandler).InstantiateDataOperator(0x59f4718, {0x3f20360, 0xc0000deb00}, 0xc00051c1b0)
	/home/runner/work/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler/oci.go:197 +0xb99
github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context.(*GadgetContext).initAndPrepareOperators(0xc0000deb00, 0xc00051c150)
	/home/runner/work/inspektor-gadget/inspektor-gadget/pkg/gadget-context/run.go:70 +0x536
github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context.(*GadgetContext).PrepareGadgetInfo(0xc0000978b8?, 0x2?)
	/home/runner/work/inspektor-gadget/inspektor-gadget/pkg/gadget-context/run.go:163 +0x13
github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local.(*Runtime).GetGadgetInfo(0x3efe618?, {0x3f31cd8, 0xc0000deb00}, 0xd?, 0xc000097dd8?)
	/home/runner/work/inspektor-gadget/inspektor-gadget/pkg/runtime/local/oci.go:26 +0x2b
github.com/inspektor-gadget/inspektor-gadget/gadgets/ci/image_inspect/test/unit.extractGadgetInfo(0x1)
	/home/runner/work/inspektor-gadget/inspektor-gadget/gadgets/ci/image_inspect/test/unit/image_inspect_test.go:234 +0xec8
github.com/inspektor-gadget/inspektor-gadget/gadgets/ci/image_inspect/test/unit.TestInspectCmd(0xc0004928c0)
	/home/runner/work/inspektor-gadget/inspektor-gadget/gadgets/ci/image_inspect/test/unit/image_inspect_test.go:60 +0x4f
testing.tRunner(0xc0004928c0, 0x3c5b0a0)
	/opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1792 +0xf4
created by testing.(*T).Run in goroutine 1
	/opt/hostedtoolcache/go/1.24.4/x64/src/testing/testing.go:1851 +0x413
```

- https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/15815195281/attempts/1#summary-44572978202
- https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/15861467368/attempts/1#summary-44719967501
- https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/15856415984/attempts/2#summary-44704132707

--- a/gadgets/ci/image_inspect/test/unit/image_inspect_test.go
+++ b/gadgets/ci/image_inspect/test/unit/image_inspect_test.go
@@ -196,13 +196,10 @@ func TestExtraInfoLeak(t *testing.T) {
 
 func extractGadgetInfo(extraInfo bool) (*api.GadgetInfo, error) {
 	image := gadgetrunner.GetGadgetImageName("ci/image_inspect")
+	ocihandlerOp := ocihandler.New()
 
 	opGlobalParams := make(map[string]*params.Params)
-	ociParams := apihelpers.ToParamDescs(ocihandler.OciHandler.InstanceParams()).ToParams()
-
-	ocihandler.OciHandler.Init(ociParams)
-
-	opGlobalParams["oci"] = apihelpers.ToParamDescs(ocihandler.OciHandler.GlobalParams()).ToParams()
+	opGlobalParams["oci"] = apihelpers.ToParamDescs(ocihandlerOp.GlobalParams()).ToParams()
 	verifyImage := "false"
 	if verifyImage == "true" || verifyImage == "false" {
 		opGlobalParams["oci"].Set("verify-image", verifyImage)
@@ -214,11 +211,11 @@ func extractGadgetInfo(extraInfo bool) (*api.GadgetInfo, error) {
 	runtimeParams := runtime.ParamDescs().ToParams()
 
 	ops := make([]operators.DataOperator, 0)
-	err := ocihandler.OciHandler.Init(opGlobalParams["oci"])
+	err := ocihandlerOp.Init(opGlobalParams["oci"])
 	if err != nil {
 		return nil, fmt.Errorf("Error initializing OCI handler: %w\n", err)
 	}
-	ops = append(ops, ocihandler.OciHandler)
+	ops = append(ops, ocihandlerOp)
 
 	gadgetCtx := gadgetcontext.New(
 		context.Background(),
@@ -228,6 +225,7 @@ func extractGadgetInfo(extraInfo bool) (*api.GadgetInfo, error) {
 		gadgetcontext.IncludeExtraInfo(extraInfo),
 	)
 
+	ociParams := apihelpers.ToParamDescs(ocihandlerOp.InstanceParams()).ToParams()
 	paramValueMap := make(map[string]string)
 	ociParams.CopyToMap(paramValueMap, "operator.oci.")
 

--- a/pkg/operators/oci-handler/oci.go
+++ b/pkg/operators/oci-handler/oci.go
@@ -62,11 +62,18 @@ type ociHandler struct {
 	globalParams *params.Params
 }
 
+func New() *ociHandler {
+	return &ociHandler{}
+}
+
 func (o *ociHandler) Name() string {
 	return "oci"
 }
 
 func (o *ociHandler) Init(params *params.Params) error {
+	if o.globalParams != nil {
+		return fmt.Errorf("ociHandler already initialized")
+	}
 	o.globalParams = params
 	return nil
 }
@@ -562,4 +569,4 @@ func (o *ociHandler) Priority() int {
 }
 
 // OciHandler is a singleton of ociHandler
-var OciHandler = &ociHandler{}
+var OciHandler = New()

--- a/pkg/testing/gadgetrunner/gadgetrunner.go
+++ b/pkg/testing/gadgetrunner/gadgetrunner.go
@@ -173,7 +173,7 @@ func (g *GadgetRunner[T]) RunGadget() {
 		gadgetOperatorOpts = append(gadgetOperatorOpts, simple.OnStart(g.onGadgetRun))
 	}
 
-	g.DataOperator = append(g.DataOperator, ocihandler.OciHandler)
+	g.DataOperator = append(g.DataOperator, ocihandler.New())
 	for _, dataOperator := range operators.GetDataOperators() {
 		g.DataOperator = append(g.DataOperator, dataOperator)
 	}

